src/main/java/dev/devature/penguin_api/controller/IssueController.java

```diff
@@ -0,0 +1,97 @@
+package dev.devature.penguin_api.controller;
+
+import dev.devature.penguin_api.entity.Issue;
+import dev.devature.penguin_api.enums.IssueResult;
+import dev.devature.penguin_api.service.IssueService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/v1/issue")
+public class IssueController {
+
+    private final IssueService issueService;
+
+    @Autowired
+    public IssueController(IssueService issueService){
+        this.issueService = issueService;
+    }
+
+    /**
+     * @param issue Take in a RequestBody with the issue object created.
+     * @return ResponseEntity with status and body based on the result of service.
+     */
+    @PostMapping("/new")
+    public ResponseEntity<String> postIssue(@RequestBody Issue issue){
+        IssueResult issueResult = issueService.createIssue(issue);
+        return stringResponseEntity(issueResult);
+    }
+
+    /**
+     * @param issue Take in a RequestBody with the issue object created.
+     * @param id Takes in a Path variable with the issue ID to be updated.
+     * @return ResponseEntity with status and body based on the result of service.
+     */
+    @PatchMapping("/{id}")
+    public ResponseEntity<String> patchIssue(@RequestBody Issue issue, @PathVariable Long id){
+        IssueResult issueResult = issueService.updateIssue(issue, id);
+        return stringResponseEntity(issueResult);
+    }
+
+    /**
+     * @param id Take in an ID as a PathVariable.
+     * @return The issue object if it request works or {@code null} if
+     * something happened.
+     */
+    @GetMapping("/{id}")
+    public ResponseEntity<Issue> getIssue(@PathVariable Long id){
+        Issue issue = issueService.getIssue(id);
+
+        if(issue == null){
+            return ResponseEntity.notFound().build();
+        }
+
+        return ResponseEntity.ok(issue);
+    }
+
```

Comment:
The suggestion to return all active issues for the User's organization doesn't align with the current design. In this system, issue management is best handled by the columns, which represent different stages or statuses within a project. Since an organization can have multiple projects, and each project may have its own set of issues, retrieving all issues for the organization would be inefficient and unnecessary.

The IssueController should be responsible solely for CUD (Create, Update, and Delete) operations, while the columns handle the propagation and retrieval of issues based on their status. As such, I will remove the getIssue method, as it doesn't serve a purpose in this design, and delegate issue retrieval to the service layer instead.

Reply:
Fair point. I think you should still leave in the `getIssue` method you've written though, as it would be useful to link to and retrieve a singular issue. But if you feel strongly towards removing it, I'm fine with that, too.